chatkit-core/src/main/kotlin/com/pusher/chatkit/state/AuxiliaryState.kt

```diff
@@ -0,0 +1,18 @@
+package com.pusher.chatkit.state
+
+import com.pusher.chatkit.rooms.state.JoinedRoomsState
+
+internal data class AuxiliaryState(
+    val version: Int,
+    val joinedRoomsReceived: ReducerLastChange<JoinedRoomsState>
+)
+
+internal data class ReducerLastChange<T>(val version: Int, val data: T)
+
+internal object AuxiliaryStateUtil {
+    internal fun applyReducerLastChangeJoinedRoomsState(state: ChatkitState, joinedRoomsState: JoinedRoomsState):
```

Comment:
This could be a generic extension `fun` on the auxiliary state: `<C> AuxiliaryState.trackLastChange(change: LastChange<C>)`. I added `C` as we need to distinguish between changes that save the same data (e.g joined vs updated room). Per reducer we would need a data class describing its state change, for joined rooms:
```kotlin
data class JoinedRoomsReceivedStateChange(change: JoinedRoomsState)
```

Then:
```kotlin
internal data class ReducerLastChange<T>(val version: Int, val data: T)
```
should then become:
```kotlin
internal data class LastChange<C>(val version: Int, val change: C)
```
Do that we always have type of the change plus the data that has changed.